### PR TITLE
Include Dependency Frameworks in set of TFMs considered for package testing

### PR DESCRIPTION
**Pull request by ericstj**
Fixes https://github.com/dotnet/runtime/issues/27503

I believe this should fix testing for meta-packages as it will consider Frameworks for testing if they are used in dependency groups (not just files).

I haven't been able to try this out in dotnet/runtime yet and would want to do so before merging.

### CODE REVIEW

File: src/Microsoft.DotNet.PackageTesting.Tests/GetCompatibilePackageTargetFrameworksTests.cs [hunk 1]
```diff
@@ -135,7 +136,7 @@ public GetCompatibilePackageTargetFrameworksTests()
         [MemberData(nameof(PackageTfmData))]
         public void GetCompatibleFrameworks(List<string> filePaths, List<NuGetFramework> expectedTestFrameworks)
         {
-            Package package = new("TestPackage", "1.0.0", filePaths);
+            Package package = new("TestPackage", "1.0.0", filePaths, Enumerable.Empty<NuGetFramework>());
```
Comment (Anipik) [inline src/Microsoft.DotNet.PackageTesting.Tests/GetCompatibilePackageTargetFrameworksTests.cs:139]: can we add a test with empty ```filepaths``` and non-empty ```dependencyFrameworks```  ?

**Comment by ericstj** [comment]
Tested locally and this does catch the windows compat bug:
```
C:\src\dotnet\runtime\artifacts\bin\testPackages\projects\Microsoft.Windows.Compatibility\project.csproj : error NU1603: Microsoft.Windows.Compatibility 6.0.0-dev depends on System.Data.SqlClient (>= 4.8.2-dev) but System.Data.SqlClient 4.8.2-dev was not found. An approximate best match of System.Data.SqlClient 4.8.2 was resolved. [C:\src\dotnet\runtime\artifacts\bin\testPackages\dirs.proj]
```

Also found a new failure:
```
C:\src\dotnet\runtime\artifacts\bin\testPackages\projects\Microsoft.Windows.Compatibility\project.csproj : error NU1603: runtime.native.System.IO.Ports 6.0.0-dev depends on runtime.linux-arm.runtime.native.System.IO.Ports (>= 6.0.0-dev) but runtime.linux-arm.runtime.native.System.IO.Ports 6.0.0-dev was not found. An approximate best match of runtime.linux-arm.runtime.native.System.IO.Ports 6.0.0-preview.1.21071.15 was resolved. [C:\src\dotnet\runtime\artifacts\bin\testPackages\dirs.proj]
C:\src\dotnet\runtime\artifacts\bin\testPackages\projects\Microsoft.Windows.Compatibility\project.csproj : error NU1603: runtime.native.System.IO.Ports 6.0.0-dev depends on runtime.linux-arm64.runtime.native.System.IO.Ports (>= 6.0.0-dev) but runtime.linux-arm64.runtime.native.System.IO.Ports 6.0.0-dev was not found. An approximate best match of runtime.linux-arm64.runtime.native.System.IO.Ports 6.0.0-preview.1.21071.15 was resolved. [C:\src\dotnet\runtime\artifacts\bin\testPackages\dirs.proj]
C:\src\dotnet\runtime\artifacts\bin\testPackages\projects\Microsoft.Windows.Compatibility\project.csproj : error NU1603: runtime.native.System.IO.Ports 6.0.0-dev depends on runtime.linux-x64.runtime.native.System.IO.Ports (>= 6.0.0-dev) but runtime.linux-x64.runtime.native.System.IO.Ports 6.0.0-dev was not found. An approximate best match of runtime.linux-x64.runtime.native.System.IO.Ports 6.0.0-preview.1.21071.15 was resolved. [C:\src\dotnet\runtime\artifacts\bin\testPackages\dirs.proj]
C:\src\dotnet\runtime\artifacts\bin\testPackages\projects\Microsoft.Windows.Compatibility\project.csproj : error NU1603: runtime.native.System.IO.Ports 6.0.0-dev depends on runtime.osx-x64.runtime.native.System.IO.Ports (>= 6.0.0-dev) but runtime.osx-x64.runtime.native.System.IO.Ports 6.0.0-dev was not found. An approximate best match of runtime.osx-x64.runtime.native.System.IO.Ports 6.0.0-preview.1.21071.15 was resolved. [C:\src\dotnet\runtime\artifacts\bin\testPackages\dirs.proj]
```
Here we'll just need to disable that project since it's expected that a single build isn't producing all these packages.

**Review by Anipik — APPROVED** [review]
LGTM, just need a test in place.

File: src/Microsoft.DotNet.PackageTesting.Tests/GetCompatibilePackageTargetFrameworksTests.cs (hunk 1 above)
Comment (ericstj) [reply src/Microsoft.DotNet.PackageTesting.Tests/GetCompatibilePackageTargetFrameworksTests.cs:139]: Added one.

**Review by safern — APPROVED** [review]
Thanks!

**Review by Anipik — APPROVED** [review]

**Comment by ericstj** [comment]
Thanks @Anipik.  I merged dotnet/runtime#55976 so we should be ready to accept this change.  It's still possible that new failures are introduced.  @missymessa let us know if you see any dotnet/runtime failures around package testing during arcade ingestion.

**Review by ViktorHofer — COMMENTED** [review]
LGTM
